emote/callbacks.py

```diff
@@ -24,20 +24,44 @@ class LoggingMixin:
     handled by a Logger.
     """
 
-    def __init__(self, cycle: int = 0):
-        super().__init__(cycle)
+    def __init__(self, *, default_window_length: int = 250, **kwargs):
+        super().__init__(**kwargs)
+
         self.scalar_logs: Dict[str, Union[float, torch.Tensor]] = {}
+        self.windowed_scalar: Dict[str, deque[Union[float, torch.Tensor]]] = {}
+        self.windowed_scalar_cumulative: Dict[str, int] = {}
         self.image_logs: Dict[str, torch.Tensor] = {}
         self.hist_logs: Dict[str, Union[float, torch.Tensor]] = {}
         self.video_logs: Dict[str, Tuple[np.ndarray, int]] = {}
 
+        self._default_window_length = default_window_length
+
     def log_scalar(self, key: str, value: Union[float, torch.Tensor]):
         """Use log_scalar to periodically log scalar data."""
         if isinstance(value, torch.Tensor):
             self.scalar_logs[key] = value.item()
         else:
             self.scalar_logs[key] = value
 
+    def log_windowed_scalar(self, key: str, value: Union[float, torch.Tensor]):
+        """Use log_windowed_scalar to periodically log scalar data."""
```

Comment:
Maybe we could mention that this actually provides a time series. With this docstring it's not clear what the difference is between the windowed_scalar and the scalar version.

Reply:
Next PR..! Just kidding. Will fix :-)